### PR TITLE
Key functionality of potion result

### PR DESCRIPTION
**Pull request by TrevorWest66**

### CODE REVIEW

File: Assets/Scripts/Adapter/Adapter.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    private PotionDictAdapter potionAdapter = new PotionDictAdapter();
+
+    public Adapter(string key, int value) : base(key, value) { }
+
+    public override void AddToDictionary()
+    {
+        potionAdapter.AddPotionToDictionary(Key, Value);
+    }
+}
```
Comment (ellie-jean) [inline Assets/Scripts/Adapter/Adapter.cs:19]: Nice use of the adapter pattern. This was a good idea!

File: Assets/Scripts/Inventory/Inventory.cs [hunk 2]
```diff
@@ -30,20 +30,29 @@ public void PopulateInventory()
             string key = keyValue.Key;
             int value = keyValue.Value;
 
-            //Load the 2d sprite image located in the Resources folder
-            theSprite = Resources.Load<Sprite>(key + " Image");
+            if (key.Substring(0,2).Equals("55"))
```
Comment (ellie-jean) [inline Assets/Scripts/Inventory/Inventory.cs]: What is the 55?

File: Assets/Scripts/Plants/CollectablePlantsEnum.cs [hunk 3]
```diff
@@ -16,15 +16,15 @@ public enum CollectablePlantsEnum
     Ginger,
     [Description("Spearmint")]
     Spearmint,
-    [Description("Lavendar")]
+    [Description("Lavender")]
     Lavender,
```
Comment (ellie-jean) [inline Assets/Scripts/Plants/CollectablePlantsEnum.cs:20]: Didn't realize I had so many misspellings .... whoops

File: Assets/Scripts/Potions/Potion.cs [hunk 4]
```diff
@@ -10,6 +10,7 @@ public class Potion
     private int potionCoinValue;
     private Sprite potionImage;
     private List<CollectablePlantsEnum> plantsInPotion;
+    public string keyName = "55";
```
Comment (ellie-jean) [inline Assets/Scripts/Potions/Potion.cs]: I see where the 55 comes from now... wondering if there is something that's more intuitive we could add?

File: Assets/Scripts/Potions/PotionCrafter.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+// 04/25/2021
+// Class that creates a potion form a list of ingrediants
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PotionCrafter
+{
+    private Potion CreatedPotion;
```
Comment (ellie-jean) [inline Assets/Scripts/Potions/PotionCrafter.cs:11]: This class is kinda lengthy, I wonder if there's a better way to go about this. We're nearing the end so it's not a huge deal.

File: Assets/Scripts/Singletons/CurrentGameObjects.cs [hunk 6]
```diff
@@ -12,6 +12,8 @@ public sealed class CurrentGameObjects
     // Lists all the objects populated on the map
     private List<GameObject> objectsPopulated = new List<GameObject>();
 
+    public List<string> Ingrediants { get; set; } = new List<string>();
```
Comment (ellie-jean) [inline Assets/Scripts/Singletons/CurrentGameObjects.cs]: Misspelling of ingredients

File: Assets/Scripts/Singletons/CurrentGameObjects.cs [hunk 7]
```diff
@@ -36,33 +38,39 @@ public static CurrentGameObjects Instance
...
+    public void RemoveObject(GameObject aPlant)
     {
         objectsPopulated.Remove(aPlant);
     }
 
-    public void removePotion(Potion aPotion)
+    public void RemovePotion(Potion aPotion)
     {
         this.potionsCrafted.Remove(aPotion);
+        objectsCollected.Remove(aPotion.keyName);
```
Comment (ellie-jean) [inline Assets/Scripts/Singletons/CurrentGameObjects.cs:64]: Isn't the key name 55 for every potion? will this remove the potion correctly?

File: Assets/Scripts/Potions/PotionCrafter.cs (hunk 5 above)
Comment (TrevorWest66) [reply Assets/Scripts/Potions/PotionCrafter.cs:11]: I felt the same way but figured its the end of the semester and doesn't relate to a pattern so just said whatever it works lol

**Review by ellie-jean — APPROVED** [review]
Looks better. I still see Ingredients misspelled in a few spots but that's not a huge deal since it's still readable. Thanks for adding your name and date to the classes!

**Pull request merged**